STYLE.md

```diff
@@ -0,0 +1,270 @@
+#Telia Company Terraform Style Guide
+
+## Introduction
+
+This guide gives developers a guidance in the coding conventions used by Telia Company  for Terraform's HashiCorp Configuration Language (HCL). Terraform allows infrastructure to be described as code, which in turns makes changes and implementations of readable in a structured way. As such, we should adhere to a style guide to ensure readable and high quality code. As Terraform utilises [HCL](https://github.com/hashicorp/hcl), you may wish to take a detailed look at its [syntax guide](https://github.com/hashicorp/hcl/blob/master/README.md#syntax) for further guidance.
+
+Inspired by [The Ruby Style Guide](https://github.com/bbatsov/ruby-style-guide) and [The Puppet Style Guide](https://docs.puppetlabs.com/guides/style_g uide.html).
+
+## Syntax
+
+- Strings are in double-quotes.
+
+### Spacing
+
+Use 2 spaces when defining resources except when defining inline policies or other inline resources.
+
+```
+resource "aws_iam_role" "iam_role" {
+  name = "${var.resource_name}-role"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+```
+
+There should be two blank lines between resource definitions.
+
+```hcl
+// bad
+variable "vpc_id" {}
+
+variable "vpc_id" {
+  description = "The VPC this security group will go in"
+  type        = "string"
+}
+
+// goo
+variable "vpc_id" {
+  description = "The VPC this security group will go in"
+  type        = "string"
+}
+
+
+variable "public_subnet_ids" {
+  description = "List of public subnet IDs"
+  type        = "list"
+  default     = []
+}
+```
+
+
+
+### Resource Block Alignment
+
+Parameter definitions in a resource block should be aligned. The `terraform fmt` command can do this for you.
+
+```hcl
+// bad
+resource "aws_security_group" "main" {
+    name = "${var.name}"
+    description = "Security Group ${var.name}"
+    vpc_id = "${var.vpc_id}"
+    tags {
+        Name = "${var.name}"
+    }
+}
+
+// good
+resource "aws_security_group" "main" {
+  name        = "${var.name}"
+  description = "Security Group ${var.name}"
+  vpc_id      = "${var.vpc_id}"
+  tags {
+    Name = "${var.name}"
+  }
+}
+```
+### Variables
+
+Variables should be provided in a `variables.tf` file at the root of your module.
+
+A description and type should be provided for each declared variable. When it makes sense, please provide a reasonable default value.
+
+## Outputs
+
+Outputs should be provided in an `outputs.tf` file at the root of your module.
+
+Attributes exported by your modules should follow the same descriptive format as variable names.
+
+## Comments
+
+Only comment what is necessary.
+
+Single line comments: `#` or `//`.
```

Comment:
I think we should prefer one or the other. To me it's confusing when they are mixed, and VS Code defaults to # if you comment out a selection (just saying ;)